### PR TITLE
Navbar Component (Web)

### PR DESCRIPTION
**Pull request by masalcedo88**
Pull request for the addition of the Navbar Web Component (mobile pending). The Navbar component iterates through the list of pre-designated routes coming from src/reducers/routerReducer.js (can make it an API call in the future if needed) and starts rendering the parent navigation links. If the component identifies that an item has children links, it will then generate a submenu with all it's children as active links, otherwise it will render the main item as an active link.
- creates: Navbar.js, NavItem.js, DropDownMenu.js and DropDownItem.js to handle route iteration and rendering of navigation links.
- creates: Navbar.scss for all navbar styling (BEM naming)

### CODE REVIEW

File: src/components/Navbar/Navbar.scss [hunk 1]
```diff
@@ -1,18 +1,158 @@
...
 }
 
-.button-text {
-  font-family: $poppins;
-  font-weight: 600;
-  font-size: 15px;
-  line-height: 22px;
-  letter-spacing: 0.03em;
+.nav--link {
+  color: #000000;
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.scss:83]: should make this a color variable

File: src/reducers/navReducer.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  switch (action.type) {
+    case SET_ACTIVE_MENU:
+      return { ...state, activeMenu: action.payload };
+    default:
+      return state;
+  }
+};
+
+const INITIAL_STATE = {
+  activeMenu: null
```
Comment (mattlaguardia) [inline src/reducers/navReducer.js:13]: Lets make this false instead of null - it will distinguish that this should be a bool value

File: src/styles/normalize.scss [hunk 3]
```diff
@@ -59,7 +59,6 @@ header,
 hgroup,
 main,
 menu,
-nav,
```
Comment (mattlaguardia) [inline src/styles/normalize.scss:62]: why was this deleted?

File: src/components/Navbar/DropDownMenu.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { setActiveMenu } from '../../actions/index';
+
+const DropDownMenu = (props) => {
+  const dispatch = useDispatch();
+  const dropdownRef = useRef(null);
+
+  const renderMenuItems = props.items.map((item) => <DropDownItem name={item.name} linkTo={item.linkTo} />);
+
+  return (
+    <div className="nav--dropdown" ref={dropdownRef} onMouseLeave={() => dispatch(setActiveMenu(null))}>
```
Comment (mattlaguardia) [inline src/components/Navbar/DropDownMenu.js:16]: I think hover issue is with how this is constructed

File: src/components/Navbar/NavItem.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      onMouseEnter={() => {
+        dispatch(setActiveMenu(props.name));
+      }}
+      to={props.children ? '#' : props.linkTo}
+    >
+      <li href="#" className={props.name === 'Donate' ? 'donate--link__txt' : 'nav--link__txt'}>
+        {props.name}
+      </li>
+      {navState.activeMenu === props.name && props.children}
+    </Link>
```
Comment (mattlaguardia) [inline src/components/Navbar/NavItem.js:25]: This logic could use some work this dispatch is always run when mouse enters, what happens where there is no children to show, this might cause a memory leak depending on how react handles on mouse enter

Comment (masalcedo88) [reply src/components/Navbar/NavItem.js:25]: I left the dispatch() on links with no children so that if in the future the client needs to add children to a nav-link our navItem component can always look at the redux store and handle the closing of the DropDownMenu when hovering across links.

File: src/components/Navbar/Navbar.js [hunk 6]
```diff
@@ -1,42 +1,47 @@
...
+  };
+
+  return (
+    <nav className="nav">
+      <div className="nav--container">
+        <Link className="nav--container" to="/">
+          <img className="nav--logo" src={logo} alt="HUOA Logo" />
+          <div className="nav--txt">
+            <h4 className="nav--txt__title">Hawaii United Okinawa Association</h4>
+            <p className="nav--txt__body">Celebrating 120 years of Uchinanchu in Hawaii</p>
```
[the remark this answers is not in the document]
Comment (masalcedo88) [reply src/components/Navbar/Navbar.js]: Mike helped me with a lil bug we had @ index.js

File: src/components/Navbar/Navbar.scss [hunk 7]
```diff
@@ -1,18 +1,158 @@
...
+.nav--container {
+  text-decoration: none;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+}
+
+.nav--logo {
+  height: $navSize * .67;
```
[the remark this answers is not in the document]
Comment (masalcedo88) [reply src/components/Navbar/Navbar.scss:30]: Listo! Got rid of all my "beautiful mind" calculations XD

File: src/assets/huoa-logo.svg [hunk 8]
```diff
@@ -0,0 +1,2584 @@
+<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="336.833px" height="342px" viewBox="398.583 238.5 336.833 342" enable-background="new 398.583 238.5 336.833 342" xml:space="preserve">
+<g>
```
Comment (mattlaguardia) [inline src/assets/huoa-logo.svg:2]: You should be able to safely remove this `<g>` tag since its just grouping

Comment (mattlaguardia) [reply src/assets/huoa-logo.svg:2]: Could you also fix the indents?

File: src/components/Navbar/DropDownItem.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+const DropDownItem = (props) => {
+  return (
+    <Link className="nav--dropdown__link" to={props.linkTo}>
+      <li>{props.name}</li>
+    </Link>
+  );
+};
+
+export default DropDownItem;
```
Comment (mattlaguardia) [inline src/components/Navbar/DropDownItem.js:12]: 🤔 idk if this should be a separate component

Comment (masalcedo88) [reply src/components/Navbar/DropDownItem.js]: true, it was too simple. refactored and deleted

File: src/components/Navbar/NavItem.js (hunk 5 above)
Comment (masalcedo88) [reply src/components/Navbar/NavItem.js]: dispatch now only fires if navlink has children

File: src/components/Navbar/DropDownMenu.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+//Components, assets, actions, styles etc..
+import { setActiveMenu } from '../../actions/index';
+
+const DropDownMenu = (props) => {
+  const dispatch = useDispatch();
+
+  const closeMenu = (e) => {
+    e.stopPropagation();
+    dispatch(setActiveMenu(false));
+  };
```
Comment (mattlaguardia) [inline src/components/Navbar/DropDownMenu.js:15]: Whats the purpose of this? I don't think you logic here works correctly

File: src/components/Navbar/NavItem.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+//Components, assets, actions, styles etc..
+import { setActiveMenu } from '../../actions/index';
+
+const NavItem = (props) => {
+  const navState = useSelector((state) => state.nav);
+  const dispatch = useDispatch();
+
+  const setMenu = () => dispatch(setActiveMenu(props.name));
+  const closeMenu = () => dispatch(setActiveMenu(false));
+  const setClass = () => (props.name === 'Donate' ? 'donate' : 'nav');
```
Comment (mattlaguardia) [inline src/components/Navbar/NavItem.js:15]: This seems like over kill with using redux - instead i think it makes more sense to just keep this to component state.

Comment (mattlaguardia) [reply src/components/Navbar/NavItem.js:15]: Example: 
```
const NavItem = (props) => {
  const [ navState, setNavState ] = useState(null);

  const setMenu = () => setNavState(props.name);
  const closeMenu = () => setNavState(null);
  const setClass = () => (props.name === 'Donate' ? 'donate' : 'nav');

  return (
    <div className={`${setClass()}--link`} onMouseEnter={setMenu} onMouseLeave={closeMenu}>
      <Link className={`${setClass()}--link__txt`} to={props.linkTo}>
        {props.name}
      </Link>
      {navState && props.children}
    </div>
  );
};
```

Comment (mattlaguardia) [reply src/components/Navbar/NavItem.js:15]: This way we don't do extra calls to the store - we dont need to mutate state here so we shouldnt be using the store to handle this state.

Comment (mattlaguardia) [reply src/components/Navbar/NavItem.js:15]: ☝️ this is a okay solution intention is to just to show functionality

File: src/components/Navbar/NavItem.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+  return (
+    <div className={`${setClass()}--link`} onMouseEnter={setMenu} onMouseLeave={closeMenu}>
```
Comment (mattlaguardia) [inline src/components/Navbar/NavItem.js:18]: I dont love dynamically setting classes like this for an loop what happens when donate is not last (some thing to think about)

File: src/components/Navbar/DropDownMenu.js (hunk 10 above)
Comment (masalcedo88) [reply src/components/Navbar/DropDownMenu.js:15]: Cleaned up!

File: src/components/Navbar/Navbar.js [hunk 13]
```diff
@@ -1,42 +1,47 @@
...
-    return (
-      <div>
-        <Link to="/">
-          <h4>HUOA Home</h4>
+const Navbar = () => {
+  const routerState = useSelector((state) => state.router);
+
+  const hasDropdown = (children) => {
+    if (children.length) {
+      return <DropDownMenu items={children} />;
```
[the remark this answers is not in the document]
Comment (masalcedo88) [reply src/components/Navbar/Navbar.js:17]: Moved DropDownMenu component into Navbar.js

File: src/components/Navbar/Navbar.js [hunk 14]
```diff
@@ -1,42 +1,47 @@
...
-          </Link>
-          <Link to="/donate">
-            <li>Donate</li>
-          </Link>
-        </ul>
       </div>
-    );
-  }
-}
+      <ul className="nav--links">{renderNavItems(routerState.routes)}</ul>
```
[the remark this answers is not in the document]
Comment (masalcedo88) [reply src/components/Navbar/Navbar.js:42]: renamed to nav--items.

this <ul> only holds parent <li>s

File: src/components/Navbar/NavItem.js (hunk 12 above)
Comment (masalcedo88) [reply src/components/Navbar/NavItem.js:18]: So the way it is currently written, the donate button would appear with the chunky purple styling regardless of where in the array it appears, even if it's not last.

File: src/components/Navbar/NavItem.js (hunk 11 above)
Comment (masalcedo88) [reply src/components/Navbar/NavItem.js]: Navbar.js has been refactored to hold component level state via react hooks and avoid calling redux store for conditional rendering of dropdown menu.

File: src/resources/font.js [hunk 15]
```diff
@@ -3,9 +3,9 @@ var WebFont = require('webfontloader');
 const fonts = () => {
   WebFont.load({
     google: {
-      families: [ 'Open Sans', 'Poppins', 'Lora' ]
+      families: [ 'Open Sans', 'Poppins: 100,300,500,600,900', 'Lora: 300,600,900' ]
```
Comment (mattlaguardia) [inline src/resources/font.js:6]: This is way too many fonts to load each time you add a number thats another font that will need to be loaded

File: src/components/Navbar/Navbar.scss [hunk 16]
```diff
@@ -1,18 +1,147 @@
...
 }
 
-.button-text {
-  font-family: $poppins;
-  font-weight: 600;
-  font-size: 15px;
-  line-height: 22px;
-  letter-spacing: 0.03em;
-  text-transform: uppercase;
+.donate--link {
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.scss:125]: This should be under `.nav` since thats its parent element - maybe a good name would be `.nav--donateLink`

File: src/components/Navbar/Navbar.js [hunk 17]
```diff
@@ -1,42 +1,60 @@
...
+          {dropdown(item.children)}
+        </NavItem>
+      );
+    });
+  };
+
+  return (
+    <nav className="nav">
+      <Link to="/">
+        <div className="nav--container">
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.js:47]: IMO container type elements should be on the outer not inner

File: src/components/Navbar/Navbar.js [hunk 18]
```diff
@@ -1,42 +1,60 @@
...
+      return (
+        <NavItem name={item.name} linkTo={item.linkTo} key={item.linkTo} navState={navState} setNavState={setNavState}>
+          {dropdown(item.children)}
+        </NavItem>
+      );
+    });
+  };
+
+  return (
+    <nav className="nav">
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.js:45]: class name of nav is probs not the greatest since nav is an element, maybe use navbar?

File: src/components/Navbar/Navbar.js [hunk 19]
```diff
@@ -1,42 +1,60 @@
...
+
+  const renderNavItems = (items) => {
+    return items.map((item) => {
+      return (
+        <NavItem name={item.name} linkTo={item.linkTo} key={item.linkTo} navState={navState} setNavState={setNavState}>
+          {dropdown(item.children)}
+        </NavItem>
+      );
+    });
+  };
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.js:42]: This could be cleaned up a bit
```suggestion
  const renderNavItems = (items) => {
    return items.map((item) => {
      return (
        <NavItem name={item.name} linkTo={item.linkTo} key={item.linkTo} navState={navState} setNavState={setNavState}>
          { !!item.children.length &&
            <div className="nav--dropdown">
              <ul className="nav--dropdown__links">{renderMenuItems(item.children)}</ul>
            </div>
          }
        </NavItem>
      );
    });
  };
```

File: src/components/Navbar/Navbar.js (hunk 17 above)
Comment (mattlaguardia) [reply src/components/Navbar/Navbar.js:47]: this is more of a wrapper class than a container

File: src/components/Navbar/Navbar.js [hunk 20]
```diff
@@ -1,42 +1,60 @@
...
-}
+const Navbar = () => {
+  const [ navState, setNavState ] = useState(false);
+  const { routes } = useSelector((state) => state.router);
+
+  const renderMenuItems = (children) => {
+    return children.map((item) => (
+      <Link className="nav--dropdown__link" to={item.linkTo} onClick={() => setNavState(false)} key={item.linkTo}>
+        <li className="nav--dropdown__link--txt">{item.name}</li>
+      </Link>
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.js:20]: Too much nesting with naming convention

why can't we just do this 
```suggestion
      <Link to={item.linkTo} onClick={() => setNavState(false)} key={item.linkTo}>
        <li className="nav--dropdown__link">{item.name}</li>
      </Link>
```

File: src/styles/typography.scss [hunk 21]
```diff
@@ -27,6 +28,20 @@ a {
...
+    font-weight: 300;
+  }
+  li {
+    font-family: $open-sans;
+  }
+}
+
+.donate--link a {
+  font-weight: 500;
+}
```
Comment (mattlaguardia) [inline src/styles/typography.scss:42]: This should be done in nav.scss not in our typography

File: src/components/Navbar/Navbar.js [hunk 22]
```diff
@@ -1,42 +1,60 @@
 //Dependencies
 import React from 'react';
 import { Link } from 'react-router-dom';
+import { useState } from 'react';
```
Comment (mcheng09) [inline src/components/Navbar/Navbar.js:4]:
```suggestion
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
```

File: src/styles/colors.scss [hunk 23]
```diff
@@ -2,12 +2,12 @@
...
 $darkest-purple: #5E264E;
 $dark-purple: #763263;
-$purple: #9F4E8D;
-$light-purple: #EFE8F6;
-$lighter-purple: #F7F0F5;
-$dark-green: #194F1A;
+$purple: #9f4e8d;
+$light-purple: #efe8f6;
+$lighter-purple: #f7f0f5;
+$dark-green: #194f1a;
```
Comment (mcheng09) [inline src/styles/colors.scss:13]: Are we doing lowercase for hex colors?

File: src/index.js [hunk 24]
```diff
@@ -23,14 +23,8 @@ const Main = (
...
-      document.getElementById('root')
-    );
+    ReactDOM.hydrate(Main, document.getElementById('root'));
   };
 } else {
-  ReactDOM.render(
-    Main,
-    document.getElementById('root')
-  );
+  ReactDOM.render(Main, document.getElementById('root'));
```
Comment (mcheng09) [inline src/index.js:29]: I don't mind this format, it's pretty.

File: src/styles/colors.scss (hunk 23 above)
Comment (mcheng09) [reply src/styles/colors.scss:13]: Please switch back to uppercase for all hex colors

File: src/components/Navbar/Navbar.scss [hunk 25]
```diff
@@ -1,18 +1,147 @@
...
+      display: flex;
+      color: $black;
+      align-items: center;
+      justify-content: center;
+      font-size: 1.1rem;
+
+      &:hover {
+        color: $dark-purple;
+        text-shadow: 0 0 .9px $dark-purple, 0 0 .9px $dark-purple, 0 0 .9px $dark-purple;
+      }
```
Comment (mcheng09) [inline src/components/Navbar/Navbar.scss:84]: Might want to add a transition effect unless it hasn't been decided.

File: src/reducers/routerReducer.js [hunk 26]
```diff
@@ -0,0 +1,122 @@
...
+        {
+          name: 'Okinawan Classes',
+          linkTo: '/events/classes'
+        }
+      ]
+    },
+    {
+      name: 'Geneology',
+      linkTo: '/geneology',
+      children: []
```
Comment (mcheng09) [inline src/reducers/routerReducer.js:91]: Is the children property needed if there is no child? You can also change your logic to just { if (children) } in your const dropdown.

File: src/components/Navbar/Navbar.js [hunk 27]
```diff
@@ -1,42 +1,60 @@
...
+  const renderMenuItems = (children) => {
+    return children.map((item) => (
+      <Link className="nav--dropdown__link" to={item.linkTo} onClick={() => setNavState(false)} key={item.linkTo}>
+        <li className="nav--dropdown__link--txt">{item.name}</li>
+      </Link>
+    ));
+  };
+
+  const dropdown = (children) => {
+    if (children.length) {
```
Comment (mcheng09) [inline src/components/Navbar/Navbar.js:25]: See reducer for new if logic

File: src/styles/colors.scss (hunk 23 above)
Comment (mcheng09) [reply src/styles/colors.scss:13]: Looks like we'll be doing uppercase! Please switch back.

File: src/components/Navbar/Navbar.js (hunk 27 above)
Comment (masalcedo88) [reply src/components/Navbar/Navbar.js]: dropdown is no longer a function, what's your take on integrating this if I'm now merging this into renderNavItems function per mateo's suggesiton?

File: .stylelintignore [hunk 28]
```diff
@@ -1 +1 @@
-src/styles/normalize.scss
+src/styles/normalize.scss
```
Comment (mattlaguardia) [inline .stylelintignore:1]: can you back this change out? IMO there should be a newline in this file

File: src/components/Navbar/NavItem.js [hunk 29]
```diff
@@ -4,15 +4,15 @@ import { Link } from 'react-router-dom';
...
 
   return (
     <div
-      className={`${setClass()}--link`}
+      className={`navbar__${setClass()}`}
       onMouseEnter={() => setNavState(name)}
       onMouseLeave={() => setNavState(false)}
     >
-      <Link className={`${setClass()}--link__txt`} to={linkTo}>
+      <Link className={`navbar__${setClass()}--txt`} to={linkTo}>
```
Comment (mattlaguardia) [inline src/components/Navbar/NavItem.js:15]: NOTE: this will change with mike's button PR

File: src/components/Navbar/Navbar.js [hunk 30]
```diff
@@ -15,45 +14,39 @@ const Navbar = () => {
 
   const renderMenuItems = (children) => {
     return children.map((item) => (
-      <Link className="nav--dropdown__link" to={item.linkTo} onClick={() => setNavState(false)} key={item.linkTo}>
-        <li className="nav--dropdown__link--txt">{item.name}</li>
+      <Link className="navbar__dropdown--link" to={item.linkTo} onClick={() => setNavState(false)} key={item.linkTo}>
+        <li className="navbar__dropdown--link__txt">{item.name}</li>
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.js:18]: this breaks bem conventions we shouldnt have a 4th
```
navbar__dropdown--txt
```

File: src/components/Navbar/Navbar.scss [hunk 31]
```diff
@@ -1,95 +1,96 @@
...
+@import '../../styles//utilities/colors.scss';
 @import '../../styles/fonts.scss';
+@import '../../styles/utilities/breakpoints.scss';
 
-.nav {
-  &--container {
-    height: 10rem;
-    background-color: $white;
+.navbar {
+  &__container {
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.scss:6]: I think @JasonToups has a helpful `<Container>` component for you to use

File: stylelint.config.js [hunk 32]
```diff
@@ -1,289 +1,295 @@
...
-				{
-					type: "at-rule",
-					name: "media",
+        'dollar-variables',
+        'custom-properties',
+        'at-rules',
+        'declarations',
+        {
+          type: 'at-rule',
+          name: 'media'
```
Comment (mattlaguardia) [inline stylelint.config.js:40]: mike pushed up changes to this file (there will be conflicts)

File: src/components/Navbar/Navbar.scss (hunk 25 above)
Comment (masalcedo88) [reply src/components/Navbar/Navbar.scss]: Jenn said they weren't using a transition effect as of 08/19

File: stylelint.config.js (hunk 32 above)
Comment (masalcedo88) [reply stylelint.config.js:40]: I'll be ready

File: src/actions/index.js [hunk 33]
```diff
@@ -6,11 +6,15 @@ const api = 'https://dev.huoa.org/wp-json';
 
 export const getPages = () => async (dispatch) => {
   const response = await axios.get(`${api}/wp/v2/pages?page=1&per_page=100`);
-  const data = response.data.reduce((allData, { slug, title, acf }) => ({ ...allData, [slug]: { title: title.rendered, ...acf }}), {});
+  const data = response.data.reduce(
+    (allData, { slug, title, acf }) => ({ ...allData, [slug]: { title: title.rendered, ...acf } }),
+    {}
+  );
```
Comment (mcheng09) [inline src/actions/index.js:12]: Is this change from the prettifier plugin?

File: src/components/Navbar/NavItem.js [hunk 34]
```diff
@@ -0,0 +1,31 @@
+// Dependencies
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+// Components, actions, assets, etc..
+import Button from '../Button/Button';
```
Comment (mcheng09) [inline src/components/Navbar/NavItem.js:6]: a bit nit picky but you can do `import Button from 'Button/Button';`.

File: src/components/Navbar/Navbar.js [hunk 35]
```diff
@@ -1,42 +1,63 @@
...
+        <NavItem name={item.name} linkTo={item.linkTo} key={item.linkTo} navState={navState} setNavState={setNavState}>
+          {!!item.children && (
+            <div className="navbar__dropdown">
+              <ul className="navbar__dropdown--links">{renderMenuItems(item.children)}</ul>
+            </div>
+          )}
+        </NavItem>
+      );
+    });
+  };
```
Comment (mcheng09) [inline src/components/Navbar/Navbar.js:41]: While it's working, it's not good practice to nest <a> within another <a>. See below for reference:

[https://stackoverflow.com/questions/9882916/are-you-allowed-to-nest-a-link-inside-of-a-link](url)

There's a few work arounds you can do. Let me know if you'd like to talk it through.

Comment (mattlaguardia) [reply src/components/Navbar/Navbar.js:41]: Good catch mike

File: package-lock.json [hunk 36]
```diff
@@ -141,6 +141,26 @@
...
+        "@babel/types": {
+          "version": "7.11.0",
+          "resolved": "https://registry.npmjs.org/@babel/types/-/types-7.11.0.tgz",
+          "integrity": "sha512-O53yME4ZZI0jO1EVGtF1ePGl0LHirG4P1ibcD80XyzZcKhcMFeCXmh4Xb1ifGBIV233Qg12x4rBfQgA+tmOukA==",
+          "requires": {
+            "@babel/helper-validator-identifier": "^7.10.4",
+            "lodash": "^4.17.19",
+            "to-fast-properties": "^2.0.0"
+          }
+        }
```
Comment (mattlaguardia) [inline package-lock.json:163]: Lets remove these updates - there was no deps added with your PR so there should be no updates in package-lock

File: src/components/Navbar/Navbar.scss [hunk 37]
```diff
@@ -0,0 +1,153 @@
...
+    font-family: $poppins;
+
+    &:hover {
+      color: $dark-purple;
+      font-weight: 900;
+    }
+
+    &::after {
+      content: attr(data-text);
+      content: attr(data-text) / "";
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.scss:91]: why is content getting called twice?

File: src/resources/font.js [hunk 38]
```diff
@@ -3,9 +3,9 @@ var WebFont = require('webfontloader');
 const fonts = () => {
   WebFont.load({
     google: {
-      families: [ 'Open Sans', 'Poppins', 'Lora' ]
+      families: [ 'Open Sans', 'Poppins: 100,300,600', 'Lora: 100,600' ]
```
[the remark this answers is not in the document]
Comment (mattlaguardia) [reply src/resources/font.js:6]: I dont think there is any use case for 100 but can you double check with design

File: stylelint.config.js [hunk 39]
```diff
@@ -1,289 +1,295 @@
 // Transform from PascalCase to kebab-case
-const pascalToKebabCase = str => str.replace(/\.?([A-Z])/g, (char) =>  "-" + char.toLowerCase()).replace(/^-/, "");
+const pascalToKebabCase = (str) => str.replace(/\.?([A-Z])/g, (char) => '-' + char.toLowerCase()).replace(/^-/, '');
```
[the remark this answers is not in the document]
Comment (mcheng09) [reply stylelint.config.js]: I removed this in the last PR for style-linter: #44 

update(style-linter): adds css combinator logic to style linter #44 

We'll want to remove it from this branch.

File: src/resources/font.js (hunk 38 above)
[the remark this answers is not in the document]
Comment (mcheng09) [reply src/resources/font.js:6]: I'll sync up with @masalcedo88 and design on this as well since I plan on reviewing the design typography soon.

File: src/components/Navbar/Navbar.js (hunk 35 above)
Comment (masalcedo88) [reply src/components/Navbar/Navbar.js:41]: Presto, I found a fix, lmk what you think

File: src/styles/utilities/breakpoints.scss [hunk 40]
```diff
@@ -1,7 +1,7 @@
 /* breakpoint variables */
 
 $breakpoint-laptop-large: 1440px;
-$breakpoint-laptop: 1200px;
+$breakpoint-laptop: 1320px;
```
Comment (mattlaguardia) [inline src/styles/utilities/breakpoints.scss:4]: lets keep the 1200px and do something like 

```
$breakpoint-laptop-medium: 1320px;
$breakpoint-laptop-small: 1200px;
```

File: src/components/Navbar/Navbar.scss [hunk 41]
```diff
@@ -0,0 +1,157 @@
...
+        }
+      }
+    }
+
+    &--links {
+      padding: 0;
+      list-style-type: none;
+    }
+  }
+}
```
Comment (mattlaguardia) [inline src/components/Navbar/Navbar.scss:157]: there is no new line at the end of this file please add one

File: src/components/Navbar/NavItem.js [hunk 42]
```diff
@@ -0,0 +1,32 @@
...
+
+// Components, actions, assets, etc..
+import Button from 'components/Button/Button';
+
+const NavItem = (props) => {
+  const { name, linkTo, children, navState, setNavState } = props;
+  const isActive = navState === name ? 'navbar__link--active' : null;
+
+  return name === 'Donate' ? (
+    <Button type="outlined" link="donate">
```
Comment (mattlaguardia) [inline src/components/Navbar/NavItem.js:13]: @mcheng09 i think marco needs to do a `/` yea?
```suggestion
    <Button type="outlined" link="/donate">
```

**Review by mattlaguardia — APPROVED** [review]

**Comment by mattlaguardia** [comment]
You need to rebase before you can merge
ALSO i left one last comment
